### DISCUSSION TITLE
Does Apollo Router(Hive) support streaming? Apollo Router as such doesnt support streaming unless its enterprise license

### DISCUSSION THREAD

**Discussion by techtreet** [graphql-hive/console#5060] [Help] [answered]
Does Apollo Router(Hive) support streaming? Apollo Router as such doesnt support streaming unless its enterprise license https://www.apollographql.com/docs/router/executing-operations/subscription-support/

This is a deal breaker for me to decide either to use guild hive or wundergraph.

**Comment by kamilkisiela** [comment, marked as answer]
@techtreet 

You can use Hive with other routers. Hive is a schema registry (not only though) that works with other gateways.

[GraphQL Mesh](https://the-guild.dev/graphql/mesh) supports Subscriptions, is MIT and works with GraphOS, GraphQL Hive and pretty much everything out there.
It also supports Apollo Federation in 100% (we tested it on A LOT of variations of a supergraph - if I recall correctly it was ~260k variations).

### What I mean by "supports Apollo Federation"?

**Gateway**
That you won't experience a situation where it fails to resolve a query or resolves it incorrectly. So pretty much **correctness**.
We see it very much missing in the conversation about gateways, they all just share numbers like p95, p99 or memory usage, but in reality these numbers do not matter if your GraphQL result is corrupted...
Most of the gateways out there work for trivial, up to semi-advanced Supergraphs, we made sure our is covering 100% of Federation.

I know that I shared no "proofs", but we do have a Federation compatibility checking tool that we will announce soon and it clearly shows who really supports and understands what Apollo Federation is capable of and what rules apply there (long story short, it's us - The Guild, and Apollo).


**Registry**
The same applies to the registries. In order to compose a supergraph, you need verify if the subgraphs are following the rules of Apollo Federation. That's why we created our own composition library (MIT - https://the-guild.dev/blog/open-source-apollo-federation) that is battle tested and works for all our users and future users.

Stay tuned for GraphQL Conf :)
